File: Mbs/Trading/Data/Generators/Square/SquareParameters.cs
using System.Collections.Generic;
using System.ComponentModel;
using System.ComponentModel.DataAnnotations;

namespace Mbs.Trading.Data.Generators.Square
{
    /// <summary>
    /// The input parameters for the square impulse generator.
    /// </summary>
    public class SquareParameters : IValidatableObject
    {
        /// <summary>
        /// Gets or sets the amplitude of the square impulse, should be positive.
        /// </summary>
        [Required(ErrorMessage = ErrorMessages.FieldIsRequired, AllowEmptyStrings = false)]
        [DefaultValue(DefaultParameterValues.SquareAmplitude)]
        [Range(0.0, double.MaxValue, ErrorMessage = ErrorMessages.FieldMustBePositive)]
        public double Amplitude { get; set; } = DefaultParameterValues.SquareAmplitude;

        /// <summary>
        /// Gets or sets the value corresponding to the minimum of square impulse, should be positive.
        /// </summary>
        [Required(ErrorMessage = ErrorMessages.FieldIsRequired, AllowEmptyStrings = false)]
        [DefaultValue(DefaultParameterValues.SquareMinimalValue)]
        [Range(0.0, double.MaxValue, ErrorMessage = ErrorMessages.FieldMustBePositive)]
        public double MinimalValue { get; set; } = DefaultParameterValues.SquareMinimalValue;

        /// <inheritdoc />
        public IEnumerable<ValidationResult> Validate(ValidationContext validationContext)
        {
            yield break;
        }
    }
}

File: Mbs.Api.Host.Ng/ClientApp/src/app/shared/mbs/data/generators/square/square-parameters.ts
import { amplitudeName, minimalValueName, objectName } from '../constants';

/** The input parameters for the square impulse generator. */
export class SquareParameters {
    private static readonly defaultAmplitude: number = 100;
    private static readonly defaultMinimalValue: number = 10;

    /** The amplitude of the square impulse, should be positive. */
    amplitude: number = SquareParameters.defaultAmplitude;

    /** The minimum of square impulse, should be positive. */
    minimalValue: number = SquareParameters.defaultMinimalValue;

    constructor(data?: SquareParameters) {
        if (data) {
            for (const property in data) {
                if (data.hasOwnProperty(property)) {
                    (<any>this)[property] = (<any>data)[property];
                }
            }
        }
    }

    toJSON(data?: any) {
        data = typeof data === objectName ? data : {};
        data[amplitudeName] = this.amplitude;
        data[minimalValueName] = this.minimalValue;
        return data;
    }
}
